pkg/distributor/http.go

```diff
@@ -37,7 +38,7 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 	default:
-		if _, err := util.ParseProtoReader(r.Context(), r.Body, &req, util.RawSnappy); err != nil {
+		if _, err := util.ParseProtoReader(r.Context(), r.Body, int(r.ContentLength), math.MaxInt32, &req, util.RawSnappy); err != nil {
```

Comment:
ParseProtoReader uses max size to create a LimitedReader.  It feels strange to pass `math.MaxInt32` in here, but reviewing the code it seems like it would be fine:

https://github.com/golang/go/blob/master/src/io/io.go#L444

Reply:
Nothing more to add here, but I'd echo @joe-elliott's comments.

Reply:
I'm not sure what to use for this value, but it seems that the limit reader is used only for another CompressionType (not `RawSnappy`).

Reply:
see https://github.com/cortexproject/cortex/blob/bc767362ef159d8960143064c224898a813decd9/pkg/util/http.go#L83

RawSnappy doesn't use a `LimitedReader`.